packages/core/cli/utils.ts

```diff
@@ -71,11 +77,60 @@ const contractExists = async (
     return code.length > 2
 }
 
+/**
+ * Retrieves and parses on-chain Unirep contract data to create an off-chain
+ * representation as a UnirepState object.
+ * @param provider An Ethereum provider
+ * @param address The address of the Unirep contract
+ */
+const genUnirepState = async (
```

Comment:
I see we have another genUnirepState in core/test/utils.ts. What is different between them?

Reply:
They're exact copies, I just don't like importing code from the test files in the production code. It's possible to import from the CLI but since it's only in 2 places I figure it's fine. Another options is putting it in a shared utils file in src that isn't exported from index.

We can't export it in the default one because the browser can't use/import `SQLiteConnector`.

Reply:
So I tried exporting the function and requiring arguments for db and prover, but it seems like you may as well write the whole function at that point. The problem is, we don't want to use `defaultProver` or `SQLiteConnector` so that in a browser, for example, they don't  have to import and then stub `path`, `sqlite`, etc.